receiver/kubeletstatsreceiver/internal/metadata/metrics.go

```diff
@@ -14,113 +14,127 @@
 
 package metadata // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/kubeletstatsreceiver/internal/metadata"
 
+import "go.opentelemetry.io/collector/pdata/pcommon"
+
+type RecordDoubleDataPointFunc func(*MetricsBuilder, pcommon.Timestamp, float64)
+
+type RecordIntDataPointFunc func(*MetricsBuilder, pcommon.Timestamp, int64)
+
+type RecordIntDataPointWithDirectionFunc func(*MetricsBuilder, pcommon.Timestamp, int64, string, AttributeDirection)
+
+type MetricsBuilders struct {
+	WithNodeStartTime    *MetricsBuilder
+	WithPodStartTime     *MetricsBuilder
+	WithDefaultStartTime *MetricsBuilder
```

Comment:
Usually with apply `With...` to optional arguments. Lest call these fields `NodeMetricsBulder`, `PodMetricsBulder` and `OtherMetricsBuilder`

Reply:
Will it be confusing when PodMetricsBuilder is used in the `containerStats` function?  Pod and Container stats both use pod's start time.

Reply:
Why will it be confusing?

Reply:
Might not be, but I wanted to make sure it was clear that in the context of a container will be doing

```golang
addCPUMetrics(a.mbs.PodMetricsBuilder, metadata.ContainerCPUMetrics, s.CPU, currentTime)
addMemoryMetrics(a.mbs.PodMetricsBuilder, metadata.ContainerMemoryMetrics, s.Memory, currentTime)
addFilesystemMetrics(a.mbs.PodMetricsBuilder, metadata.ContainerFilesystemMetrics, s.Rootfs, currentTime)
```

Reply:
this looks correct to me

Reply:
Rename pushed